apollo-federation/src/merger/merger.rs

```diff
@@ -1186,8 +1187,79 @@ impl Merger {
         Ok(source_type.clone())
     }
 
-    pub(in crate::merger) fn merge_description<T>(&mut self, _sources: &Sources<T>, _dest: &T) {
-        todo!("Implement merge_description")
+    pub(in crate::merger) fn merge_description<T>(&mut self, sources: &Sources<T>, dest: &T)
+    where
+        T: HasDescriptionPosition + std::fmt::Display,
+    {
+        let mut descriptions: Vec<String> = Vec::new();
+        let mut counts: Vec<isize> = Vec::new();
+
+        for source in sources.values() {
+            if source.is_none() || source.as_ref().unwrap().description(&self.merged).is_none() {
+                continue;
+            }
+
+            let source_desc = source
+                .as_ref()
+                .unwrap()
+                .description(&self.merged)
+                .unwrap()
+                .as_str();
+            let idx = descriptions.iter().position(|desc| desc == source_desc);
+            if let Some(idx) = idx {
+                counts[idx] += 1;
+            } else {
+                // Very much a hack but simple enough: while we do merge 'empty-string' description if that's all we have (debatable behavior in the first place,
+                // but graphQL-js does print such description and fed 1 has historically merged them so ...), we really don't want to favor those if we
+                // have any non-empty description, even if we have more empty ones across subgraphs. So we use a super-negative base count if the description
+                // is empty so that our `indexOfMax` below never pick them if there is a choice.
+                if source_desc.is_empty() {
+                    counts.push(isize::MIN);
+                } else {
+                    counts.push(1);
+                }
+                descriptions.push(source_desc.to_string());
+            }
+        }
+
+        if !descriptions.is_empty() {
+            // we don't want to raise a hint if a description is ""
+            let non_empty_descriptions = descriptions
+                .iter()
+                .filter(|desc| !desc.is_empty())
+                .collect_vec();
+            if descriptions.len() == 1 {
```

Comment:
Just a suggestion: I'd like to use `iter_into_single_item` in this case.
```suggestion
            if let Some(only_description) = iter_into_single_item(descriptions.iter()) {
```